### PR TITLE
Offline mode integration

### PR DESCRIPTION
**Pull request by janm399**
Hopefully the last of the offline mode requests. This completes the integration with the mobile, allowing the iOS client to send the offline mode request in one block, and to also provide a mechanism for the client to indicate that it should move to offline mode if there is an inconsistency in the data.

Merging this PR will break the old serialized ``MultiPacket``s, and it changes the decoder to make it incompatible with the previous protocol.

The ``MultiPacket`` now includes ``timestamp: Long``, which represents time in milliseconds when the multipacket was constructed. The timestamp does not need to have any specific starting point (e.g. UNIX epoch), but it needs to be monotonously increasing with passing time for the duration of the session. On the server side, it is possible for the ``PacketWithLocation`` to contain zero-length ``BitVector``, which indicates that the sensor at the given location did not provide any data, even though it was expected.

* [x] Server replay haldler
* [x] iOS protocol change to handle timestamp and missing data
* [x] Server protocol change to handle timestamp and missing data

### CODE REVIEW

File: server/exercise/src/main/scala/com/eigengo/lift/exercise/MultiPacket.scala [hunk 1]
```diff
@@ -4,10 +4,10 @@ import scodec.bits.BitVector
 
 case class PacketWithLocation(sourceLocation: SensorDataSourceLocation, payload: BitVector)
 
-case class MultiPacket(packets: List[PacketWithLocation]) {
+case class MultiPacket(timestamp: Long, packets: List[PacketWithLocation]) {
```
Comment (carlpulley) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/MultiPacket.scala]: I understand the reason for transmitting a long value on the wire, however, is there any utility in decoding this value into a semantically meaningful timestamp value (e.g. a Jodatime DateTime)?

Comment (carlpulley) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/MultiPacket.scala]: Sorry, just read the last packet in the PR notes - so scrubbed the previous comment!

File: server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesProcessor.scala [hunk 2]
```diff
@@ -264,17 +293,24 @@ class UserExercisesProcessor(override val kafka: ActorRef, notification: ActorRe
...
-      val fos = new FileOutputStream(s"/Users/janmachacek/$id.mp")
-      fos.write(data)
-      fos.close()
+    case ExerciseSessionReplayProcessData(`newSessionId`, data) ⇒
+      persist(Seq(SessionAbandonedEvt(oldSessionId), SessionStartedEvt(newSessionId, sessionProps))) { case _ :: started :: Nil ⇒
+        log.warning("Not yet handling processing of the data")
+        sender() ! \/.right(())
+
+        // TODO: Implement proper replay handling. For now, we save the file and end the session.
+        val fos = new FileOutputStream(s"/Users/janmachacek/$newSessionId.mp")
```
Comment (carlpulley) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesProcessor.scala]: Shouldn't we pull home directory here from an environment variable or similar?

Comment (janm399) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesProcessor.scala]: Indeed we should.

**Comment by carlpulley** [comment]
:+1:

**Comment by janm399** [comment]
Done. Next up, #46

**Pull request merged**